src/model.js

```diff
@@ -0,0 +1,242 @@
+  const sort = setSort(options);
+  if (sort) {
+    query.sort_by = sort;
+  }
+
+  query.limit = 50;
+  return query;
+}
+
+/**
+ * Given a bounding box (assuming web mercator right now), finds the center
+ * and returns [longitude, latitude]
+ *
+ * @param {object} bbox
+ */
+function getCenter(bbox) {
+  const x = (bbox.xmax + bbox.xmin) / 2;
+  const y = (bbox.ymax + bbox.ymin) / 2;
+
+  return proj.forward([x, y]);
+}
+
+/**
+ * Returns distance in meters from center to outside corner of bounding box.
+ *
+ * @param {object} bbox
+ */
+function getRadius(bbox) {
+  const halfX = Math.abs(bbox.xmax - bbox.xmin);
+  const halfY = Math.abs(bbox.ymax - bbox.ymin);
+
+  const radius = Math.sqrt(Math.pow(halfX, 2) + Math.pow(halfY, 2));
+  let roundedRadius = Math.round(radius);
+  if (roundedRadius > YELP_API_MAX_RADIUS) {
+    console.log(
+      "Warning: Radius was larger than the allowed maximum from Yelp API."
+    );
+    roundedRadius = YELP_API_MAX_RADIUS;
+  }
+  return roundedRadius;
+}
+
+// This function parses the search term from a geoservices where clause
+// 'where term = 'restaurants' => restaurants
+function setTerm(options) {
+  if (!options.where) return null;
+  const match = options.where.match(/.+\s*=\s*'(.+)'/);
+  return match ? match[1] : null;
+}
+
+// This function handles the sort part of yelps query. Depending on the type of search we are doing
+// we may want to sort by distance, rating or best match
+function setSort(options) {
+  if (options.orderByFields) {
+    // there may be multiple sorts - we only assume the first one since yelp can only do one at a time.
+    let orderByFields = options.orderByFields.split(",")[0];
+
+    if (orderByFields.indexOf(" ASC") > -1) {
+      // Yelp API does not allow sorting by ascending order ("ASC").
+      // If "ASC" order is passed, do nothing.
+      return false;
+    } else {
+      // assume "DESC" sort, but remove it from the string:
+      orderByFields = orderByFields.split(" DESC")[0];
+      if (orderByFields === "best_match") {
+        return "best_match";
+      } else if (orderByFields === "rating") {
+        return "rating";
+      } else if (orderByFields === "review_count") {
+        return "review_count";
+      } else if (orderByFields === "distance") {
+        return "distance";
+      } else {
+        // order by some other column, do nothing.
+        return false;
+      }
+    }
+  } else {
+    return false;
+  }
+}
+
+// Map across all elements from a Yelp response and translate it into a feature collection
+function translate(data, query) {
+  // protect ourself in case the request did not return any features
+  if (data.businesses) {
+    return data.businesses.map((business) => {
+      return formatFeature(business, query);
+    });
+  } else {
+    console.error("NO FEATURES");
+  }
+}
+
+// This function takes a single element from the yelp response and translates it to GeoJSON
+function formatFeature(biz, query) {
```

Comment:
Please don't use slang like `biz` or `loc` - I know it's less keystrokes, but it take longer to read and over the lifetime of the repo quick readability probably more import. So use `business` and `location`.  Then use destructuring to get the properties your need.

```
const { 
  location,
  coordinates: { latitude, longitude },
  id,
  alias,
  // etc
} = business
```

Also If all you are using from the `query` parmeter is  `term`,  just pass in `term`.